### PR TITLE
chore: replace `enum` with `object as const`

### PR DESCRIPTION
**Pull request by keita-determined**
## Description
[WEB-534]

## Test Plan
- Check if all are replaced with `as const`
- Check of everything works as expected

## Commentary (optional)
- Type is more strict now which is amazing!
- I left [these enums](https://github.com/determined-ai/determined/blob/3f7196841e1183e19a9f671bca44126aca83c572/webui/react/src/shared/utils/service.test.ts#L2) as it is on purpose cuz these mock `api.ts`

## Checklist
- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

[WEB-534]: https://determinedai.atlassian.net/browse/WEB-534?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/components/Badge.tsx [hunk 1]
```diff
@@ -10,12 +10,14 @@ import { ResourceState, RunState, SlotState } from 'types';
...
-  State = 'State',
-}
+export const BadgeType = {
+  Default: 'Default',
+  Header: 'Header',
+  Id: 'Id',
+  State: 'State',
+} as const;
+
+export type BadgeType = typeof BadgeType[keyof typeof BadgeType];
```
Comment (ClaireNeveu) [inline webui/react/src/components/Badge.tsx:20]: Can we add a `ValueOf` utility type since we're now using this pattern so often? It just makes the code easier to read.

File: webui/react/src/utils/browser.ts [hunk 2]
```diff
@@ -106,7 +106,7 @@ const updateFavicon = (iconPath: string): void => {
   linkEl.href = iconPath;
 };
 
-export const updateFaviconType = (active: boolean, branding = BrandingType.Determined): void => {
+export const updateFaviconType = (active: boolean, branding: BrandingType): void => {
```
Comment (ClaireNeveu) [inline webui/react/src/utils/browser.ts:109]: You removed a default argument here, was that intentional?

File: webui/react/src/utils/task.ts [hunk 3]
```diff
@@ -182,7 +182,7 @@ export const isExperimentTask = (task: Type.AnyTask): task is Type.ExperimentTas
 export const isTaskKillable = (task: Type.AnyTask | Type.ExperimentItem): boolean => {
   return (
     killableRunStates.includes(task.state as Type.RunState) ||
-    killableCommandStates.includes(task.state as Type.CommandState)
```
Comment (ClaireNeveu) [inline webui/react/src/utils/task.ts:185]: What's the reason for this change?

File: webui/react/src/utils/browser.ts (hunk 2 above)
Comment (keita-determined) [reply webui/react/src/utils/browser.ts:109]: yes, it is used in only a place and value is set

File: webui/react/src/themes.ts [hunk 4]
```diff
@@ -23,11 +23,13 @@ import {
  * Where did we get our sizes from?
  * https://www.quora.com/What-is-the-difference-among-big-large-huge-enormous-and-giant
  */
-export enum ShirtSize {
-  small = 'small',
-  medium = 'medium',
-  large = 'large',
-}
+export const ShirtSize = {
+  small: 'small',
```
Comment (ClaireNeveu) [inline webui/react/src/themes.ts:27]: This and a couple of other enums don't follow our `TitleCase` naming convention, would you mind fixing that?

File: webui/react/src/utils/task.ts (hunk 3 above)
Comment (keita-determined) [reply webui/react/src/utils/task.ts:185]: I got type error cuz `task.state` doesnt satisfies `killableCommandStates`
we can do `map().some()` alternatively

File: webui/react/src/themes.ts (hunk 4 above)
Comment (keita-determined) [reply webui/react/src/themes.ts:27]: sure

File: webui/react/src/components/Badge.tsx (hunk 1 above)
Comment (keita-determined) [reply webui/react/src/components/Badge.tsx:20]: make sense

File: webui/react/src/utils/task.ts (hunk 3 above)
Comment (ClaireNeveu) [reply webui/react/src/utils/task.ts:185]: You should fix the type of `killableCommandStates` then, it should be an `Array<CommandState>`.

Comment (keita-determined) [reply webui/react/src/utils/task.ts]: fixed

**Review by ClaireNeveu — APPROVED** [review]
